shared/util/set.tsx

```diff
@@ -0,0 +1,3 @@
+export function intersect<T>(a: Set<T>, b: Set<T>): Set<T> {
+  return new Set([...a].filter(x => b.has(x)))
```

Comment:
```suggestion
  return new Set(a.size > b.size ? [...b].filter(x => a.has(x)) : [...a].filter(x => b.has(x)))
```

Reply:
ok but honestly i'm sure we could do a set of a million items and not have any difference here